api/tests/mocha/controllers/changes.js

```diff
@@ -2110,37 +2109,9 @@ describe('Changes controller', () => {
 
       return nextTick().then(() => {
         authorization.getAllowedDocIds.callCount.should.equal(3);
-        controller._logWarnings();
-        logger.warn.callCount.should.equal(2);
-        logger.warn.args[0].should.deep.equal(['User "userB" replicates "10500" docs']);
-        logger.warn.args[1].should.deep.equal(['User "userC" replicates "15000" docs']);
-      });
-    });
-
-    it('should not spam the logs', () => {
-      authorization.getAllowedDocIds
-        .withArgs(sinon.match({ id: 'A' })).resolves(Array.from(Array(7500), (a, i) => i))
-        .withArgs(sinon.match({ id: 'B' })).resolves(Array.from(Array(10500), (a, i) => i))
-        .withArgs(sinon.match({ id: 'C' })).resolves(Array.from(Array(15000), (a, i) => i));
-
-      controller.request(testReqA, testResA);
-      controller.request(testReqA, testResA);
-      controller.request(testReqB, testResB);
-      controller.request(testReqB, testResB);
-      controller.request(testReqB, testResB);
-      controller.request(testReqC, testResC);
-      controller.request(testReqC, testResC);
-      controller.request(testReqC, testResC);
-
-      return nextTick().then(() => {
-        authorization.getAllowedDocIds.callCount.should.equal(8);
-        logger.warn.callCount.should.equal(0);
-        controller._logWarnings();
-        logger.warn.callCount.should.equal(2);
-        logger.warn.args[0].should.deep.equal(['User "userB" replicates "10500" docs']);
-        logger.warn.args[1].should.deep.equal(['User "userC" replicates "15000" docs']);
-        controller._logWarnings();
-        logger.warn.callCount.should.equal(2);
+        logDocs.logReplicationLimitExceeded.callCount.should.equal(2);
+        logDocs.logReplicationLimitExceeded.args[0].should.deep.equal(['userB', 10500, 10000]);
+        logDocs.logReplicationLimitExceeded.args[1].should.deep.equal(['userC', 15000, 10000]);
```

Comment:
You've removed the test called "should not spam the logs" but it might be worth having something somewhere to make sure we don't keep updating the doc with no (or few) changes. The easiest solution is after the doc get, only update if it's old or the count is significantly different.

Reply:
Is 1 month old or 100 difference in counting okay? 🤔

Reply:
Sounds good to me. We can tweak it if we need more accurate numbers in future.

Reply:
Fixed. The service is checking this and it's being tested in the service's spec.